Modules/Scripted/WebServer/WebServerLib/base.py

```diff
@@ -0,0 +1,73 @@
+"""
+Base interface(s) for the Slicer WebServer module.
+"""
+
+import abc
+from typing import Callable, Optional
+
+
+BaseRequestLoggingFunction = Callable[[list[any]], None]
+"""Function signature for an external handle for message logging."""
+
+
+class SlicerBaseRequestHandler(abc.ABC):
```

Comment:
This is great, but here I think the class should be BaseRequestHandler since we use Slicer in the derived class to indicate it responds to Slicer requests.